src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import React, { FC, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserFromId(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserFromId(todo.userId),
+}));
+
+let lastId = Math.max(...todos
+  .map(({ id }) => id));
+
+export const App: FC = () => {
+  const [displayedTodos, setDisplayedTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const getNewTodo = () => {
+    const newId = lastId + 1;
+
+    lastId += 1;
+
+    return {
+      id: newId,
+      title,
+      completed: false,
+      userId: +selectedUserId,
+      user: getUserFromId(+selectedUserId),
```

Comment:
better abstract it into variable
and get used to explicit data type conversion:
- `Number(...)` instead of `+...`
```
      const userId = Number(selectedUserId);
...
      userId,
      user: getUserFromId(userId),
```